app/jobs/update_ticket_links_job.rb

```diff
@@ -59,9 +84,12 @@ def post_error_status(full_repo_name, sha)
     CommitStatus.new(full_repo_name: full_repo_name, sha: sha).error
   end
 
-  def commit_on_master?(full_repo_name, sha)
-    git_repo = GitRepositoryLoader.from_rails_config.load(full_repo_name.split('/')[1], update_repo: true)
+  def extract_ticket_key_from_branch_name(branch_name)
+    /(?<ticket_key>[A-Z]{2,8}-[1-9][0-9]*)/ =~ branch_name
```

Comment:
Our project key limit is actually 10 characters, which is the default. As it's configurable in the Jira instance we might want to increase it or make it configurable. For now making it 10 instead of 8 should be fine?

Make it case insensitive too?
```suggestion
    /(?<ticket_key>[A-Z]{2,10}-[1-9][0-9]*)/i =~ branch_name
```

Reply:
Updated to 10 👍.

I was thinking that making it case sensitive would give devs a way to be clearer that a ticket key is indeed a ticket key, since if it's lower case allowed the following branch would be seen as containing a ticket: `some-feature-123`. What do you think?

Reply:
Yep makes sense, upper case is fine 👍